extensions/ql-vscode/src/packaging.ts

```diff
@@ -0,0 +1,144 @@
+import { CodeQLCliServer } from './cli';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import {
+  getOnDiskWorkspaceFolders,
+  showAndLogErrorMessage,
+  showAndLogInformationMessage,
+} from './helpers';
+import { window } from 'vscode';
+import { ProgressCallback } from './commandRunner';
+
+const CORE_PACKS = [
+  'codeql/cpp-all',
+  'codeql/csharp-all',
+  'codeql/go-all',
+  'codeql/java-all',
+  'codeql/javascript-all',
+  'codeql/python-all',
+  'codeql/ruby-all',
+];
+
+/**
+ * Lists all workspace folders that contain a qlpack.yml file.
+ *
+ * Note: This currently only finds packs at the root of a workspace folder.
+ * TODO: Add support for packs in subfolders.
+ */
+function getWorkspacePacks(): string[] {
+  const packs: string[] = [];
+  const workspaceFolders = getOnDiskWorkspaceFolders();
+  for (const folder of workspaceFolders) {
+    const qlpackYml = path.join(folder, 'qlpack.yml');
+    if (fs.pathExistsSync(qlpackYml)) {
+      packs.push(folder);
+    }
+  }
+  return packs;
+}
```

Comment:
This is my (bad) attempt to list all packs in the workspace. Currently only looks for `qlpack.yml` files in the root of the workspace, which excludes anything in the `ql` and `codeql-go` submodules in the starter workspace... Does anyone have a suggestion for how to find packs in subfolders? 

My immediate hacky thought was to hard-code the paths to the main packs in the starter workspace, but surely there's a nicer way 😅